packages/devtools_app/lib/src/shared/config_specific/drag_and_drop/drag_and_drop.dart

```diff
@@ -109,23 +118,42 @@ class DragAndDrop extends StatefulWidget {
 
 class DragAndDropState extends State<DragAndDrop> {
   final _dragging = ValueNotifier<bool>(false);
+  DragAndDropManager? _dragAndDropManager;
 
   bool _isActive = false;
 
+  void _refreshDragAndDropManager(int viewId) {
+    if (_dragAndDropManager != null) {
+      final oldViewId = _dragAndDropManager!.viewId;
+
+      // Already registered to the right drag and drop manager, so do nothing.
+      if (oldViewId == viewId) return;
+
+      _dragAndDropManager?.unregisterDragAndDrop(this);
+      _dragAndDropManager = null;
+    }
+
+    _dragAndDropManager = DragAndDropManager.getInstance(viewId);
+    _dragAndDropManager!.registerDragAndDrop(this);
+  }
+
   @override
   void initState() {
     super.initState();
-    DragAndDropManager.instance.registerDragAndDrop(this);
   }
 
   @override
   void dispose() {
-    DragAndDropManager.instance.unregisterDragAndDrop(this);
+    _dragAndDropManager?.unregisterDragAndDrop(this);
     super.dispose();
   }
 
   @override
   Widget build(BuildContext context) {
+    // Each time the widget is rebuilt it may be in a a new view. So the
+    // dragAndDropManager is refreshed to ensure that we are registered in the
+    // right context.
+    _refreshDragAndDropManager(View.of(context).viewId as int);
```

Comment:
viewId should already be typed to int, we recently changed that. Are you still on an old version of Flutter?